src/dataframerow/utils.jl

```diff
@@ -268,31 +262,25 @@ function group_rows(df::AbstractDataFrame, hash::Bool = true, sort::Bool = false
     rperm = Vector{Int}(undef, length(groups))
     copyto!(stops, starts)
     @inbounds for (i, gix) in enumerate(groups)
-        rperm[stops[gix]] = i
-        stops[gix] += 1
+        rperm[stops[gix+1]] = i
+        stops[gix+1] += 1
     end
     stops .-= 1
 
-    # drop group 1 which contains rows with missings in grouping columns
-    if skipmissing
-        popfirst!(starts)
-        popfirst!(stops)
-        groups .-= 1
-        ngroups -= 1
-    end
+    # When skipmissing=true was used, group 0 corresponds to missings to drop
+    # Otherwise it's empty
```

Comment:
maybe we can put an `@assert` for this statement - it should not be expensive I think.

Reply:
The problem is, we don't know whether `skipmissing=true` was used in this function. We could store it in the `GroupedDataFrame` just for checking, but normally tests would catch bugs in that regard anyway.

Reply:
OK